pkg/acsengine/k8s_versions.go

```diff
@@ -521,3 +521,26 @@ var KubeConfigs = map[string]map[string]string{
 		"gclowthreshold":             strconv.Itoa(DefaultKubernetesGCLowThreshold),
 	},
 }
+
+var dashboardUsesHTTP = map[string]bool{
+	common.KubernetesVersion1Dot8Dot2:  true,
+	common.KubernetesVersion1Dot8Dot1:  true,
+	common.KubernetesVersion1Dot8Dot0:  true,
+	common.KubernetesVersion1Dot7Dot10: true,
+	common.KubernetesVersion1Dot7Dot9:  true,
+	common.KubernetesVersion1Dot7Dot7:  true,
+	common.KubernetesVersion1Dot7Dot5:  true,
+	common.KubernetesVersion1Dot7Dot4:  true,
+	common.KubernetesVersion1Dot7Dot2:  true,
+	common.KubernetesVersion1Dot7Dot1:  true,
+	common.KubernetesVersion1Dot7Dot0:  true,
+	common.KubernetesVersion1Dot6Dot13: true,
+	common.KubernetesVersion1Dot6Dot12: true,
+	common.KubernetesVersion1Dot6Dot11: true,
+	common.KubernetesVersion1Dot6Dot9:  true,
+	common.KubernetesVersion1Dot6Dot6:  true,
+}
+
+func kubernetesDashboardUsesHTTP(version string) bool {
```

Comment:
Shouldn't this be based on the dashboard spec instead of the Kubernetes version?  The kubernetes versions will support the new dashboard, right?

Reply:
(Changes here will be updated since https://github.com/Azure/acs-engine/pull/1946 introduces new approach for locating addons manifest.)

Yes, actually it should be decided by dashboard image version. 
One thing to note is in https://github.com/kubernetes/kubernetes/pull/53046, it changes default redirect of `kubectl proxy` '/ui' to use https. And this change now applies for v1.9.0+ .

That means if we're using v1.8 k8s with dashboard configured to HTTPS. Navigating to '/ui' would still redirect to the HTTP dashboard, which is not available. This break existing user experience who is using '/ui' for redirecting.

Thus I'd propose enabling it for v1.9 only. See also below versioning test comment.